### PR TITLE
refactor: simplify ThreeDotsMenu component

### PR DESCRIPTION
**Pull request by rudolfs**
Remove unused functionality to reduce dependencies.

Refs https://github.com/radicle-dev/radicle-upstream/issues/1877.

Screenshot 2021-10-20 at 12 15 06

### CODE REVIEW

File: ui/DesignSystem/ThreeDotsMenu.svelte [hunk 1]
```diff
@@ -64,110 +48,89 @@
...
-                    <p>{item.title}</p>
-                  </div>
-                </Tooltip>
-              {/if}
-            {/each}
-          </div>
-        {/if}
+    {#if expanded && menuItems.length > 0}
+      <div
+        out:fade={{ duration: 100 }}
```
Comment (geigerzaehler) [inline ui/DesignSystem/ThreeDotsMenu.svelte:113]: Should we use `fade|local`?

File: ui/DesignSystem/ThreeDotsMenu.svelte [hunk 2]
```diff
@@ -64,110 +48,89 @@
...
+        class="modal"
+        hidden={!expanded}
```
Comment (geigerzaehler) [inline ui/DesignSystem/ThreeDotsMenu.svelte:115]: I believe we can drop this attribute

File: ui/DesignSystem/ThreeDotsMenu.svelte [hunk 3]
```diff
@@ -64,110 +48,89 @@
...
+              class="item"
+              class:disabled={item.disabled}
+              on:click={!item.disabled
+                ? () => {
+                    expanded = false;
+                    item.event();
+                  }
+                : undefined}>
+              <svelte:component this={item.icon} style="margin-right: 12px" />
+              <p>{item.title}</p>
```
Comment (geigerzaehler) [inline ui/DesignSystem/ThreeDotsMenu.svelte:130]:
```suggestion
              <div>{item.title}</div>
```

File: ui/DesignSystem/ThreeDotsMenu.svelte [hunk 4]
```diff
@@ -64,110 +48,89 @@
...
+    cursor: pointer;
+    margin-top: 1rem;
+    position: absolute;
+    right: 0;
+    top: 100%;
     user-select: none;
+    width: 240px;
+    z-index: 10;
+    /* Round corners of child elements. */
+    overflow: hidden;
```
Comment (geigerzaehler) [inline ui/DesignSystem/ThreeDotsMenu.svelte:73]: Do you have any idea why this doesn’t hide the tooltip that is a child element?

File: ui/DesignSystem/ThreeDotsMenu.svelte [hunk 5]
```diff
@@ -6,53 +6,37 @@
...
+    icon: typeof Icon;
     event: () => void;
     tooltip?: string;
     dataCy?: string;
     disabled?: boolean;
   }
-  export let menuItems: MenuItem[];
-  export let headerTitle: string | undefined = undefined;
 
+  import { fade } from "svelte/transition";
```
Comment (geigerzaehler) [inline ui/DesignSystem/ThreeDotsMenu.svelte:18]: The imports should precede the `interface` definition.

Comment (rudolfs) [reply ui/DesignSystem/ThreeDotsMenu.svelte:10]: 73727843fdc9ef0a6dd257ab0ae5b4eb865f95f9

File: ui/DesignSystem/ThreeDotsMenu.svelte (hunk 3 above)
Comment (rudolfs) [reply ui/DesignSystem/ThreeDotsMenu.svelte]: 2b131936a4d6144d1ea1fe888e84cc5156c9cf6d

**Review by geigerzaehler — APPROVED** [review]

**Pull request merged**